### PR TITLE
Student points dashboard

### PR DESCRIPTION
**Pull request by coreycartercodes**
## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please describe):

## Problem/feature
Add total student points calculation and integrate into index(dashboard)
- Had to add a new endpoint for allpoints (calls all student courses)

## Solution
.reduce from activerecord course_students call

## Checklist
- [X] Code compiles correctly
- [X] Tests for the changes have been added (for bug fixes/features)
- [X] All tests passing
- [ ] No rubocop offenses
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: app/controllers/students/courses_controller.rb [hunk 1]
```diff
@@ -5,47 +5,40 @@ def conn(uri)
...
-    #   Course.new(course_data)
-    # end
+    student_course_params = ({
+      student_id: @student.id
+    })
+    response = conn("/api/v1/students/courses/points").get do |request|
+      request.body = student_course_params
+    end
+    point_data = JSON.parse(response.body, symbolize_names: true)[:data]
+    @points = point_data.reduce(0) do |total, course|
```
Comment (priyapower) [inline app/controllers/students/courses_controller.rb:27]: EPIC! Love this use of reduce to quickly sum all student points!

File: app/controllers/teacher/courses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Teacher::CoursesController < ApplicationController
```
Comment (priyapower) [inline app/controllers/teacher/courses_controller.rb:1]: This file will be deleted

File: app/controllers/students/courses_controller.rb [hunk 3]
```diff
@@ -5,47 +5,39 @@ def conn(uri)
...
 
-    # response = Faraday.get("BACKEND URI")
-    # courses_data = JSON.parse(response.body, symbolize_names: true)
-    # @courses = courses_data[:data].map do |course_data|
-    #   Course.new(course_data)
-    # end
+    student_course_params = ({
+      student_id: @student.id
+    })
+    response = conn("/api/v1/students/courses/all-points").get do |request|
```
Comment (priyapower) [inline app/controllers/students/courses_controller.rb:23]: Added an extra endpoint for all points on backend (in pr #47)

File: spec/features/students/courses/points_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    response = Faraday.get('http://localhost:3000/api/v1/students/courses/points') do |request|
+      request.body = joins_params
+    end
+    my_student_course = JSON.parse(response.body, symbolize_names: true)
+    within '#my-points' do
+      expect(page).to have_content(my_student_course[:data][0][:attributes][:student_points])
+    end
+  end
+
+  scenario 'a student can see their total points on their dashboard (course index)' do
```
Comment (priyapower) [inline spec/features/students/courses/points_spec.rb:40]: Added this test for confirming all courses assigned to a student can tally up points and be visible on dashboard

**Review by priyapower — APPROVED** [review]
I added a spec and needed to make updates to both be endpoints and the controller call in FE (the updated BE branch is called student_dash_points and is in pr #47)

**Review by ckccameron — APPROVED** [review]

**Review by Arique1104 — APPROVED** [review]

**Review by sarynm12 — APPROVED** [review]

**Pull request merged**